barretenberg/cpp/src/barretenberg/commitment_schemes/utils/shplemini_accumulator.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+#include <vector>
+
+namespace bb {
+/**
+ * @brief An accumulator consisting of the Shplonk evaluation challenge and vectors of commitments and scalars.
+ *
+ * @details This structure is used in the `reduce_verify_shplemini_accumulator` method of KZG or IPA. It allows a chosen
+ * Polynomial Commitment Scheme (PCS) to perform a single batch multiplication operation by:
+ * 1. Receiving the last commitment \f$ W \f$ from the prover.
+ * 2. Adding this commitment to the vector of commitments.
+ * 3. Adding \f$ -\text{evaluation\_point} \f$ to the vector of scalars.
+ * 4. Performing a single `batch_mul` operation to obtain the pair of pairing points.
+ *
+ * @tparam Curve: BN254 or Grumpkin.
+ */
+template <typename Curve> struct ShpleminiAccumulator {
```

Comment:
I think "accumulator" is not quite right as a term for this thing, which is not some value that's being built up over multiple iterations of a single computation. It feels a bit clearer to call it somehing like `BatchOpeningClaim` or `SinglePointBatchOpeningClaim` to echo `OpeningClaim` which is already in use.